RELEASE.md

```diff
@@ -14,28 +14,54 @@
    example, in the `v0.2.0` release, the container image location is
    `us.gcr.io/k8s-artifacts-prod/capi-openstack/capi-openstack-controller:v0.2.0`
 
+## Version number
+
+A release version string is: `vX.Y.Z`.
+
+A pre-release version string additionally has a suffix:
+- `alpha` for an alpha release
+- `beta` for a beta release
+- `rc` for a release candidate
+and an additional index starting at 0. This takes the form: `vX.Y.Z-[suffix].N`. e.g. the first release candidate prior
+to version 1.2.3 would be called `v1.2.3-rc.0`.
+
+It is recommended to create at least one release candidate when bumping `X` or `Y`.
+
 ## Process
 
-1. Make sure your repo is clean by git's standards.
-1. Make sure you are on the correct branch (`main` for the current release and `release-0.x` for older releases).
+1. Make sure your repo is clean by git's standards. It is recommended to use a fresh checkout.
+1. When bumping `X` or `Y` (but not Z or the pre-release suffix) in the release version you must create a new release branch called `release-X.Y`.
+   > NOTE: `upstream` should be the name of the remote pointing to `github.com/kubernetes-sigs/cluster-api-provider-openstack`
+    - git checkout main
+    - git pull
+    - git checkout -b release-`X`.`Y`
+    - git push --set-upstream `upstream`
+1. When bumping `X` or `Y` (but not Z or the pre-release suffix) in the release version, ensure you have added a new
+   entry to [metadata.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/metadata.yaml)
+   as [described in the CAPI book](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#metadata-yaml), and
+   that this has been committed to the release branch prior to release.
+1. Make sure you are on the correct release branch: `release-X.Y`
 1. Set an environment variable with the version, e.g.:
-    - `VERSION=v0.4.0`
+    - `VERSION=v0.6.0`
 1. Create an annotated tag
     - `git tag -s -a $VERSION -m $VERSION`.
 1. Push the tag to the GitHub repository:
    > NOTE: `upstream` should be the name of the remote pointing to `github.com/kubernetes-sigs/cluster-api-provider-openstack`
     - `git push upstream $VERSION`
-1. Run `make release` to build artifacts (the image is automatically built by CI)
```

Comment:
Shouldn't we recommend producing the artifacts and release notes while waiting for the image to build?

Reply:
This process is actually already doing that, although you've reminded me that I was confused about which step actually kicks off the build. I'm going document that the image build was kicked off automatically when we pushed the tag.

Reply:
Step 7 triggers the image build, while in step 8 we expect to have the image ready. We could do plenty of other things while the image is building. I was suggesting we move step 8 to a later stage, it could even be the last step before publishing the release IIUC.
Not really a big deal though as it doesn't take that long to build the image, and engineers will likely know when they can skip ahead.

Reply:
when I do previous releases I actually wait for the image ready and create promote then notify folks to review
and in the mean time do the other release stuff concurrently